### PR TITLE
Feature/api 10/datasource

### PR DESCRIPTION
**Pull request by shotokan**
models and setup script #10

### CODE REVIEW

File: api/models/event.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+'use strict'
+
+const bcrypt = require('bcrypt');
+const bcrypt_p = require('bcrypt-promise');
```
Comment (mishelashala) [inline api/models/event.js]: non-model-related variables should be in `camelCase`

Comment (shotokan) [reply api/models/event.js]: ok, also I need to pass them to the user model

File: api/models/user.js [hunk 2]
```diff
@@ -25,5 +26,31 @@ module.exports = (sequelize, DataTypes) => {
       allowNull: false
     },
   });
+  model.beforeSave(async (user, options) => {
+    if (user.changed('password')) {
+      let hash;
```
Comment (mishelashala) [inline api/models/user.js]: you can avoid this declaration

```js
user.password = await bcrypt.hash(user.password, 12)
```

File: api/models/user.js [hunk 3]
```diff
@@ -25,5 +26,31 @@ module.exports = (sequelize, DataTypes) => {
...
+    }
+  });
+  Model.prototype.comparePassword = async function (pw) {
+    let pass;
+    try {
+      if (!this.password) throw Error('password not set');
+
+      pass = await bcrypt.compare(pw, this.password);
+
+      if (!pass) throw Error("Invalid password.");
```
Comment (mishelashala) [inline api/models/user.js]: avoid `one-line` `if` statements

favor:
```js
if (!pass) {
  throw Error('Invalid password')
}
```

File: api/models/user.js [hunk 4]
```diff
@@ -25,5 +26,31 @@ module.exports = (sequelize, DataTypes) => {
...
+        user.password = hash;
+      } catch (err) {
+        throw err;
+      }
+    }
+  });
+  Model.prototype.comparePassword = async function (pw) {
+    let pass;
+    try {
+      if (!this.password) throw Error('password not set');
```
Comment (mishelashala) [inline api/models/user.js]: avoid `one-line` `if` statements

favor:
```js
if (!this.password) {
  throw Error('password not set')
}
```

Comment (shotokan) [reply api/models/user.js]: done

File: api/models/user.js (hunk 3 above)
Comment (shotokan) [reply api/models/user.js]: done

File: api/models/user.js (hunk 2 above)
Comment (shotokan) [reply api/models/user.js]: done

**Pull request merged**